execution/scan/literal_selector.go

```diff
@@ -7,13 +7,23 @@ import (
 	"context"
 	"fmt"
 	"sync"
+	"time"
 
 	"github.com/thanos-io/promql-engine/execution/model"
 	"github.com/thanos-io/promql-engine/query"
 
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	"github.com/prometheus/prometheus/model/labels"
 )
 
+var (
+	constantMetric = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "operator_duration_seconds",
+		Help: "Time taken by each operator",
+	}, []string{"operator", "duration"})
+)
+
```

Comment:
This metric can be removed now I think, saw we are doing this in operator interface?